src/datetime.ts

```diff
@@ -0,0 +1,161 @@
+export interface DateTimeLike {
+  readonly year: number;
+  readonly month: number;
+  readonly day: number;
+  readonly hour: number;
+  readonly minute: number;
+  readonly second: number;
+  /**
+   * Whether the date and time is in UTC.
+   */
+  readonly utc: boolean;
+}
+
+/**
+ * Represents a date and time. Either local or UTC.
+ */
+export class DateTime implements DateTimeLike {
+  private readonly numeric: number;
+
+  public get year(): number {
+    return Math.floor(this.numeric / 100000000000);
+  }
+
+  public get month(): number {
+    return Math.floor((this.numeric / 1000000000) % 100);
+  }
+
+  public get day(): number {
+    return Math.floor((this.numeric / 10000000) % 100);
+  }
+
+  public get hour(): number {
+    return Math.floor((this.numeric / 100000) % 100);
+  }
+
+  public get minute(): number {
+    return Math.floor((this.numeric / 1000) % 100);
+  }
+
+  public get second(): number {
+    return Math.floor((this.numeric / 10) % 100);
+  }
+
+  public get utc(): boolean {
+    return this.numeric % 10 == 1;
+  }
+
+  private constructor(numeric: number) {
+    this.numeric = numeric;
+  }
+
+  /**
+   * Converts DateTime into Date object:
+   * - If DateTime is UTC, returns a Date object created from UTC time.
+   * - If DateTime is local, returns a Date object with local time.
+   */
+  public toDate(): Date {
+    if (this.utc) {
+      return new Date(
+        Date.UTC(
+          this.year,
+          this.month - 1,
+          this.day,
+          this.hour,
+          this.minute,
+          this.second,
+        ),
+      );
+    } else {
+      return new Date(
+        this.year,
+        this.month - 1,
+        this.day,
+        this.hour,
+        this.minute,
+        this.second,
+      );
+    }
+  }
+
+  /**
+   * Converts DateTime into a plain object.
+   */
+  public toObject(): DateTimeLike {
```

Comment:
The original idea behind this was to make it compatible with Luxon. With this it wouldn't be possible to construct a RRule date time from a luxon date time anymore (because of the `utc` property). Are there any other use-cases of this function?

However, I think it makes sense not to have a direct RRule-DateTime<->Luxon-DateTime conversion because with the new model, the conversion only makes sense in combination with the RRuleSet's `tzid`. So maybe it makes sense to have a Luxon conversion on `RRuleSet` instead (as a follow-up)?

Reply:
> The original idea behind this was to make it compatible with Luxon. With this it wouldn't be possible to construct a RRule date time from a luxon date time anymore (because of the utc property). Are there any other use-cases of this function?

You are right. I thought that type-compatibility between `toObject` and luxon's `fromObject` will be enough, but in reality it fails because of `utc` property presence.

The orignial idea was to have simple conversion option to popular date libraries. But looks like all conversion can now be done using `toDate()` method. But let's still keep this methods for serialization/deserialization purposes.

> However, I think it makes sense not to have a direct RRule-DateTime<->Luxon-DateTime conversion because with the new model, the conversion only makes sense in combination with the RRuleSet's tzid. So maybe it makes sense to have a Luxon conversion on RRuleSet instead (as a follow-up)?

I don't want to make this library explicitly depend on luxon.

But anyway now It can be converted using `luxon.DateTime.fromJSDate(dateTime.toDate(), {zone: dateTime.utc ? 'utc' : undefined})`